web/remote/nextcloud.go

```diff
@@ -0,0 +1,55 @@
+package remote
+
+import (
+	"net/http"
+
+	"github.com/cozy/cozy-stack/model/nextcloud"
+	"github.com/cozy/cozy-stack/model/permission"
+	"github.com/cozy/cozy-stack/pkg/consts"
+	"github.com/cozy/cozy-stack/pkg/jsonapi"
+	"github.com/cozy/cozy-stack/pkg/webdav"
+	"github.com/cozy/cozy-stack/web/middlewares"
+	"github.com/labstack/echo/v4"
+)
+
+func nextcloudPut(c echo.Context) error {
+	inst := middlewares.GetInstance(c)
+	if err := middlewares.AllowWholeType(c, permission.PUT, consts.Files); err != nil {
+		return err
+	}
+
+	accountID := c.Param("account")
+	nc, err := nextcloud.New(inst, accountID)
+	if err != nil {
+		return wrapNextcloudErrors(err)
+	}
+
+	path := c.Param("*")
+	if err := nc.Mkdir(path); err != nil {
+		return wrapNextcloudErrors(err)
+	}
+	return c.JSON(http.StatusCreated, echo.Map{"ok": true})
+}
+
+func nextcloudRoutes(router *echo.Group) {
+	group := router.Group("/nextcloud/:account")
+	group.PUT("/*", nextcloudPut)
```

Comment:
It's more precise. `POST` has no semantics in HTTP. `PUT` must be idem-potent, and if we do a `PUT` on a resource, the `GET` on this resource must return the same thing that the body of the `PUT` (it may be in a different format/representation). Those two conditions are met here, so it's better to use `PUT` than `POST`.

Reply:
I didn't know that! Thanks for the explanation :)